src/org/apache/cocoon/generation/UnzipGenerator.java

```diff
@@ -0,0 +1,131 @@
+package org.apache.cocoon.generation;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.avalon.framework.parameters.Parameters;
+import org.apache.cocoon.ProcessingException;
+import org.apache.cocoon.components.source.SourceUtil;
+import org.apache.cocoon.environment.SourceResolver;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.excalibur.source.Source;
+import org.apache.excalibur.source.SourceException;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
+
+/**
+ * Unzip a file.

```

Comment:
Unzip, en dan? Hoe is dit anders dan een generator met het jar: protocol?

Reply:
Ik geloof dat het verschil was dat deze het bestand daadwerkelijk uitpakt. Handig voor het bestand van de Raad voor de Rechtspraak, een zip met zips. Daar kom je nieet ver met het jar: protocol dacht ik.